sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -398,7 +398,7 @@ class FileBasedDataSourceSuite extends QueryTest with SharedSQLContext with Befo
               .contains(errorMessage(format)))
 
             msg = intercept[AnalysisException] {
-              spark.udf.register("testType", () => new NullData())
+              spark.udf.register("testType", udf(() => new NullData()).asNondeterministic())
```

Comment:
It would be good to add a comment as pointed out at [here](https://github.com/apache/spark/pull/24593/files#r283741849).

Reply:
Thanks @kiszk for your comment.   With the latest changes to use the optimization disabled by default, we don't need any changes to the existing tests.